### PR TITLE
OTP Flow Added

### PR DESCRIPTION
**Pull request by muktanshumishra24**
# Pull Request Template

Resolves or continues [AR-7719](...).

## Changes

Added OTP Login Flow on Dev Dashboard.

## Checklist

- [x] The branch name follows the format: `developer/AR-XXX-issue-name`.
- [x] The changes have been tested locally.

[AR-7719]: https://team-1624093970686.atlassian.net/browse/AR-7719?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/utils/validation.ts [hunk 1]
```diff
@@ -42,4 +42,9 @@ function isValidUrl(url: string) {
   return URLRegex.test(url)
 }
 
-export { isValidEmail, isValidUrl }
+export const useValidator = () => {
+  return {
+    email: isValidEmail,
+    url: isValidUrl,
+  }
+}
```
Comment (shrinathprabhu) [inline src/utils/validation.ts:50]: Why this change?

File: src/components/OTPInput.vue [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+</script>
+
+<template>
+  <AppOverlay>
+    <div
+      class="max-w-[360px] w-screen bg-[#1e1f1e] rounded-[10px] border-1 border-jet flex flex-col relative p-[1.75rem] gap-4"
+      role="dialog"
+    >
+      <button class="absolute right-4 top-4" @click="emit('dismiss')">
+        <img src="../assets/close.svg" alt="Close" />
```
Comment (shrinathprabhu) [inline src/components/OTPInput.vue:115]: `@/assets/close.svg`

File: src/components/AppOverlay.vue [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<template>
+  <div
+    class="fixed inset-0 before:content-{''} before:absolute before:inset-0 z-[99999] before:z-[99998] before:bg-black before:opacity-[0.75] h-[100vh] flex justify-center items-center backdrop-blur-[20px]"
```
Comment (shrinathprabhu) [inline src/components/AppOverlay.vue:3]: `h-[100vh]` = `h-screen`
`opacity-[0.75]` = `opacity-75`

Add the backdrop-blur-[20px] in tailwind config
```
theme: {
extend: {
backdropBlur: {
'xar-sm': 20px
}
}
}
```

And then use it like `backdrop-blur-xar-sm`
Use this for all the custom styles not available in tailwind.

File: src/components/OTPInput.vue [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+async function resendOTP() {
+  emit('resend')
+  startOTPCounterTimer()
+}
+</script>
+
+<template>
+  <AppOverlay>
+    <div
+      class="max-w-[360px] w-screen bg-[#1e1f1e] rounded-[10px] border-1 border-jet flex flex-col relative p-[1.75rem] gap-4"
```
Comment (shrinathprabhu) [inline src/components/OTPInput.vue:111]: Add all custom styles to tailwind config theme extend and use it here.

`p-[1.75rem]` = `p-7`

File: src/components/OTPInput.vue [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+      role="dialog"
+    >
+      <button class="absolute right-4 top-4" @click="emit('dismiss')">
+        <img src="../assets/close.svg" alt="Close" />
+      </button>
+      <div class="flex flex-col items-center justify-center gap-3 text-center">
+        <h2 class="text-[2rem] font-bold">Verification</h2>
```
Comment (shrinathprabhu) [inline src/components/OTPInput.vue:118]: Add this to tailwind config

File: src/components/OTPInput.vue [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+            ref="otpInputs"
+            v-model="otp[i - 1]"
+            type="number"
+            step="1"
+            min="0"
+            max="9"
+            pattern="\d*"
+            maxlength="1"
+            autocomplete="off"
+            class="outline-none border-none rounded-md p-2 bg-[#313030] placeholder:text-[#989898] flex flex-grow justify-center items-center text-center"
```
Comment (shrinathprabhu) [inline src/components/OTPInput.vue:142]: Use tailwind config

File: src/utils/validation.ts (hunk 1 above)
Comment (muktanshumishra24) [reply src/utils/validation.ts:50]: This is the Updated Validator Hook throughout other codes, and being used in the newer implementation of login.

**Review by shrinathprabhu — APPROVED** [review]
